nerfstudio/process_data/realitycapture_utils.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 The Nerfstudio Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helper utils for processing polycam data into the nerfstudio format."""
+
+import csv
+import json
+from pathlib import Path
+from typing import List
+
+import numpy as np
+from PIL import Image
+from rich.console import Console
+
+from nerfstudio.process_data.process_data_utils import CAMERA_MODELS
+
+CONSOLE = Console(width=120)
+
+
+def realitycapture_to_json(
+    image_filename_map: List[Path],
+    csv_filename: Path,
+    output_dir: Path,
+) -> List[str]:
+    """Convert RealityCapture data into a nerfstudio dataset.
+
+    Args:
+        image_filenames: List of paths to the original images.
+        csv_filename: Path to the csv file containing the camera poses.
+        output_dir: Path to the output directory.
+        verbose: Whether to print verbose output.
+
+    Returns:
+        Summary of the conversion.
+    """
+    data = {}
+    data["camera_model"] = CAMERA_MODELS["perspective"].value
+    # Needs to be a string for camera_utils.auto_orient_and_center_poses
+    data["orientation_override"] = "none"
+
+    frames = []
+
+    with open(csv_filename, encoding="UTF-8") as file:
+        reader = csv.DictReader(file)
+        cameras = {}
+        for row in reader:
+            for column, value in row.items():
+                cameras.setdefault(column, []).append(value)
+
+    img = np.array(Image.open(output_dir / image_filename_map[cameras["#name"][0].split(".")[0]]))
+    height, width, _ = img.shape
+
+    data["h"] = int(height)
+    data["w"] = int(width)
+
+    for i, name in enumerate(cameras["#name"]):
+        frame = {}
+        frame["file_path"] = image_filename_map[name.split(".")[0]].as_posix()
+        frame["fl_x"] = float(cameras["f"][i]) * max(width, height) / 36
+        frame["fl_y"] = float(cameras["f"][i]) * max(width, height) / 36
+        # TODO: Unclear how to get the principal point from RealityCapture, here a guess...
+        frame["cx"] = float(cameras["px"][i]) / 36.0 + width / 2.0
+        frame["cy"] = float(cameras["py"][i]) / 36.0 + height / 2.0
+        # TODO: Not sure if RealityCapture uses this distortion model
+        frame["k1"] = cameras["k1"][i]
+        frame["k2"] = cameras["k2"][i]
+        frame["k3"] = cameras["k3"][i]
+        frame["k4"] = cameras["k4"][i]
+        frame["p1"] = cameras["t1"][i]
+        frame["p2"] = cameras["t2"][i]
```

Comment:
@machenmusik Does this help or hurt the reconstruction?

Reply:
My private dataset and CottageMiniature both yield nice results with things as they are now, so at the moment it does not hurt.  Not sure I can say whether they help definitively.